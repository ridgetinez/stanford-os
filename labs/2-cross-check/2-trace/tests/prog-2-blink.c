// rewritten 2-blink.c from 1-gpio
#include "rpi.h"
#include "trace.h"

void notmain(void) {
    enum { led1 = 20, led2 = 21 };

    uart_init();
    trace_start(0);

    gpio_set_output(led1);
    gpio_set_output(led2);

    for(int i = 0; i < 10; i++) {
        gpio_set_on(led1);
        gpio_set_off(led2);
        delay_cycles(1000000);
        gpio_set_off(led1);
        gpio_set_on(led2);
        delay_cycles(1000000);
    }
    trace_stop();
    clean_reboot();
}
