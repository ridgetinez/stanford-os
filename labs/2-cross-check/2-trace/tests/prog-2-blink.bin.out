TRACE:GET32(0x20200008)=0x0
TRACE:PUT32(0x20200008)=0x1
TRACE:GET32(0x20200008)=0x1
TRACE:PUT32(0x20200008)=0x9
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
TRACE:PUT32(0x2020001c)=0x100000
TRACE:PUT32(0x20200028)=0x200000
TRACE:PUT32(0x20200028)=0x100000
TRACE:PUT32(0x2020001c)=0x200000
